src/app/authentication/reset-password.component.html

```diff
@@ -1,27 +1,36 @@
 
-    <div class="alert alert-danger" *ngIf="errorMessage">{{errorMessage}}</div>
+  <div class="alert alert-danger" *ngIf="errorMessage">{{errorMessage}}</div>
     <div class="alert alert-success" *ngIf="successMessage">{{successMessage}}</div>
 
-    <form [formGroup]="resetPasswordForm" (ngSubmit)="onSubmit(resetPasswordForm.value)">
-        <div class="form-group input-group" [ngClass]="{'has-error':!resetPasswordForm.controls['password'].valid && resetPasswordForm.controls['password'].touched}">
-            <span class="input-group-addon"><i class="glyphicon glyphicon-lock"></i></span>
-            <input type="password" class="form-control" name="password" placeholder={{language?.placeHolders.password}} [formControl]="resetPasswordForm.controls['password']">
-        </div>
-        <div class="form-control-feedback">
-            <p *ngIf="resetPasswordForm.controls['password'].hasError('required') && (resetPasswordForm.controls['password'].touched)">{{authMessages?.errors.passwordReq}}</p>
-            <p *ngIf="resetPasswordForm.controls['password'].hasError('pattern') && (resetPasswordForm.controls['password'].touched)">{{authMessages?.errors.passwordInvalid}}</p>
-        </div>
-        <div class="form-group input-group" [ngClass]="{'has-error':!resetPasswordForm.controls['confirmPassword'].valid && resetPasswordForm.controls['confirmPassword'].dirty}">
-            <span class="input-group-addon"><i class="glyphicon glyphicon-lock"></i></span>
-            <input type="password" class="form-control" name={{language?.placeHolders.confirmPassword}} placeholder="Confirm password" [formControl]="resetPasswordForm.controls['confirmPassword']">
-        </div>
-        <div class="alert alert-danger" *ngIf="resetPasswordForm.controls['confirmPassword'].hasError('MatchPassword') && resetPasswordForm.controls['confirmPassword'].dirty">{{authMessages?.errors.passwordMatch}}</div>
-        <div class="form-group">
-            <button type="submit" [disabled]="!resetPasswordForm.valid" class="btn button-branding-blue btn-lg">{{language?.messages.buttonMessage}}</button>
-        </div>
+    <form [formGroup]="resetPasswordForm" (ngSubmit)="onSubmit(resetPasswordForm.value)" id="resetForm">
+      <div class="form-container">
+        <mat-form-field appearance="standard">
+          <mat-icon class="toggleViewPassword" matSuffix (click)="hide = !hide">{{hide ? 'visibility' : 'visibility_off'}}</mat-icon>
+            <input matInput type="password" [type]="hide ? 'password' : 'text'" name="password" placeholder={{language?.placeHolders.password}} [formControl]="resetPasswordForm.controls['password']">
+          <mat-error *ngIf="resetPasswordForm.controls['password'].hasError('required')">{{authMessages?.errors.passwordReq}}</mat-error>
+          <mat-error *ngIf="resetPasswordForm.controls['password'].hasError('pattern')">{{authMessages?.errors.passwordPattern}}</mat-error>
```

Comment:
minlength for password has to be added here right?

Reply:
Done.